src/lib.rs

```diff
@@ -431,6 +432,87 @@ impl RustSemsimian {
             metric.to_string(),
         )
     }
+
+    pub fn associations_search(
+        &self,
+        object_closure_predicates: &HashSet<TermID>,
+        object_set: &HashSet<TermID>,
+        include_similarity_object: bool,
+        // _sort_by_similarity: bool,
+        // _property_filter: Option<HashMap<String, String>>,
+        // _subject_closure_predicates: Option<Vec<TermID>>,
+        // _predicate_closure_predicates: Option<Vec<TermID>>,
+        subject_set: &Option<HashSet<TermID>>,
+        subject_prefixes: &Option<Vec<TermID>>,
+        // _method: Option<String>,
+        limit: Option<usize>,
+    ) -> Vec<(f64, Option<TermsetPairwiseSimilarity>, TermID)> {
+        let db_path = RESOURCE_PATH.lock().unwrap().clone();
```

Comment:
Does this need to be fully cloned from the beginning or can it just be a ref, since it is only ever used as a ref below?

Reply:
I think I tried using a reference and it didn't behave as expected.

Reply:
It was a ref earlier that resulted in the application just stalling. I do not have an explanation for this but this works for now.

Reply:
Figured out a way to avoid cloning!